### ISSUE TITLE
Elements with role region must be given an accessible name

### ISSUE THREAD

**Issue by seanforyou23** [patternfly/patternfly#2079]
In an effort to improve our accessible experience (AX) I think we should require that users supply a proper label when using a `.pf-c-page__main-section`. 

Why? These elements represent major sections of an application, "landmarks" as they are referred to by assistive technology. Here's an example of what we get out of the box. Notice `banner` and `main` are referring to the aria _role_, and not any name that we've explicitly designated to those elements.

<img width="1680" alt="landmarks" src="https://user-images.githubusercontent.com/5942899/61479518-05296980-a962-11e9-900c-b4b35a36cdf4.png">

Now, notice how simply decorating the main sections with a proper label can give users who rely on screen readers a much better picture of the application content at a high-level.

<img width="1680" alt="labelled-landmarks" src="https://user-images.githubusercontent.com/5942899/61479744-7537ef80-a962-11e9-8b1f-61360a3f705f.png">

This stems from a [W3C Recommendation](https://www.w3.org/TR/wai-aria-1.1/#region) on how to best use the region role.

> Authors MUST give each element with role region a brief label that describes the purpose of the content in the region. Authors SHOULD reference a visible label with aria-labelledby if a visible label is present.

So really, _any_ section element (or any element with `role="region"`) should carry an accessible name, that said, I think starting with just `.pf-c-page__main-section` elements would be a great step forward.

Heads up @jgiardino @andybraren @srambach - lemme know if you have any thoughts on this!

**Comment by rachael-phillips** [comment]
@jgiardino what are your thoughts on this issue?

**Comment by jgiardino** [comment]
> This stems from a [W3C Recommendation](https://www.w3.org/TR/wai-aria-1.1/#region) on how to best use the region role.

That's a really great catch!

I actually would suggest a slightly alternate approach. In the example you show, the `.pf-c-page__main-section` elements are really there for decorative purposes, in my opinion. In other words, the text "Main Title" and the card view are contents that are related and belong together. I think it would be more confusing to identify these chunks as two separate sections on the page. It could probably also be argued that the first `<section>` should be a `<header>` element instead.

I also would not suggest providing an `aria-label` for the main region. This adds extra burden on consumers to provide a string that is localized, and also increases the risk of inconsistency (i.e. consumers might provide a different string, translators could translate it differently). Since there can only be one `<main>` element on the page, I expect that being announced as the 'main' region would be sufficient, and having this be consistently announced across all apps is going to be much more beneficial for users than the extra description. 

Also, as a general rule of thumb, I'd like us to be cautious about over labelling things, especially if similar information can be derived from the html elements being used, like the structure of html elements or use of headings. My concern with too many labels is that it can create noise and cognitive burden for some users.    

For this issue, I would suggest the following updates:
- Update the Page component to not use `<section>` elements. 
- Maybe add documentation to the Page component to explain that the `.pf-c-page__main-section` elements could use more semantic html elements than what we show in the example, but these elements should be based on what contents are included inside these elements. And maybe provide a link to external documentation that explains when to use different landmark roles. And also include the note about aria-label if using `<section>` elements for these.
- Update all Page demos to use html elements that are semantically correct for the contents they include. And if `<section>` is correct, to include an `aria-label` as @seanforyou23 suggests.
